snuba/subscriptions/executor_consumer.py

```diff
@@ -124,10 +124,27 @@ def join(self, timeout: Optional[float] = None) -> None:
 
 
 class SubscriptionExecutorProcessingFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(
+        self, dataset: Dataset, max_concurrent_queries: int, metrics: MetricsBackend
+    ) -> None:
+        self.__dataset = dataset
+        self.__max_concurrent_queries = max_concurrent_queries
+        self.__metrics = metrics
+
     def create(
         self, commit: Callable[[Mapping[Partition, Position]], None]
     ) -> ProcessingStrategy[KafkaPayload]:
-        return Noop(commit)
+        executor = ThreadPoolExecutor(self.__max_concurrent_queries)
```

Comment:
You do not need to create a threadpool each time you create the step, which can happen often. Please keep it static.

Reply:
Moved this line out of the factory and passing it in now so it doesn't get recreated on partition assignment, etc